### PR TITLE
Fixes #12829 - add identifier for smart variables/parameters

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

File: app/views/puppetclasses/_class_parameters.html.erb [hunk 1]
```diff
@@ -15,6 +15,9 @@
   <tr id="puppetclass_<%= puppetclass.id %>_params[<%= lookup_key.id %>]"
     class="fields <%= 'overridden' if overridden %>">
     <%= content_tag(:td, puppetclass.name, :rowspan => lookup_keys.size, :class => "ellipsis") if index == 0 %>
+    <td class="ellipsis param_name">
+      <%= lookup_key.type == "PuppetclassLookupKey" ? "Smart Parameters":"Smart Variables"%>
```
Comment (tbrisker) [inline app/views/puppetclasses/_class_parameters.html.erb]: Please extract the strings for translation (https://theforeman.org/handbook.html#Stringsandtranslations).
You can also check the key type using `lookup_key.is_a? PuppetclassLookupKey` (reducing one string allocation per key)

File: app/views/puppetclasses/_class_parameters.html.erb [hunk 2]
```diff
@@ -15,6 +15,9 @@
   <tr id="puppetclass_<%= puppetclass.id %>_params[<%= lookup_key.id %>]"
     class="fields <%= 'overridden' if overridden %>">
     <%= content_tag(:td, puppetclass.name, :rowspan => lookup_keys.size, :class => "ellipsis") if index == 0 %>
+    <td class="ellipsis param_name">
```
Comment (tbrisker) [inline app/views/puppetclasses/_class_parameters.html.erb]: I think `param_name` class isn't needed here and might actually lead to issues (iirc, it is used by the js to lookup the param name)

File: app/views/puppetclasses/_classes_parameters.html.erb [hunk 3]
```diff
@@ -2,6 +2,7 @@
   <thead class="white-header">
     <tr>
       <th class='col-md-3'><%= _('Puppet Class') %></th>
+      <th class='col-md-2'><%= _('Type') %></th>
```
Comment (tbrisker) [inline app/views/puppetclasses/_classes_parameters.html.erb]: This width causes the total column to go over 12 which is the default grid size used in bootstrap. Perhaps the width of other columns should be adjusted to fit this in.

**Review by tbrisker — CHANGES_REQUESTED** [review]
See comments inline. thanks!

File: app/views/puppetclasses/_class_parameters.html.erb [hunk 4]
```diff
@@ -16,6 +16,9 @@
     class="fields <%= 'overridden' if overridden %>">
     <%= content_tag(:td, puppetclass.name, :rowspan => lookup_keys.size, :class => "ellipsis") if index == 0 %>
     <td class="ellipsis param_name">
```
Comment (tbrisker) [inline app/views/puppetclasses/_class_parameters.html.erb]: `param_name` class should be on the other column that contains the param name (lookup_key.key). Both columns should have the `ellipsis` class so they get a tooltip if they overflow (e.g. on a small display). I think the order should be the other way around, first name then type.

File: app/views/puppetclasses/_class_parameters.html.erb [hunk 5]
```diff
@@ -16,6 +16,9 @@
...
+      <%= (lookup_key.is_a? PuppetclassLookupKey) ? _("Smart Parameters"):_("Smart Variables")%>
```
Comment (tbrisker) [inline app/views/puppetclasses/_class_parameters.html.erb]: I think these should be singular. also, please add a space around the `:`

**Review by tbrisker — CHANGES_REQUESTED** [review]

**Comment by shiramax** [comment]
@tbrisker Done.

File: app/views/puppetclasses/_class_parameters.html.erb [hunk 6]
```diff
@@ -18,6 +18,9 @@
     <td class="ellipsis param_name">
       <%= lookup_key.key %>
     </td>
+    <td class="ellipsis">
+      <%= (lookup_key.is_a? PuppetclassLookupKey) ? _("Smart Parameter"): _("Smart Variables")%>
+    <td>
```
Comment (tbrisker) [inline app/views/puppetclasses/_class_parameters.html.erb]: This should be `</td>`

**Review by tbrisker — CHANGES_REQUESTED** [review]

**Comment by shiramax** [comment]
@tbrisker  Done.

**Review by tbrisker — APPROVED** [review]

**Comment by tbrisker** [comment]
Thanks @shiramax !

**Pull request merged**